### PR TITLE
Added JDK implementation

### PR DESCRIPTION
**Pull request by lukas-krecan**
It uses HttpServer which is part of the JDK. The advantage is that we do not get version clash
which may easily happen with Jetty

### CODE REVIEW

File: RELEASENOTES.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Release notes
 
+## 1.1.2
```
Comment (liry) [inline RELEASENOTES.md]: I would raise minor version (by semantic versioning).

File: jadler-jdk/pom.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>net.jadler</groupId>
+        <artifactId>jadler-pom</artifactId>
+        <version>1.1.2-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>jadler-jdk</artifactId>
+    <version>1.1.2-SNAPSHOT</version>
```
Comment (liry) [inline jadler-jdk/pom.xml]: Remove version to use one from parent. Also provide other elements like scm, name, url, etc. (see pom in other some module).

File: jadler-jdk/src/main/java/net/jadler/stubbing/server/jdk/JdkStubHttpServer.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2012 - 2015 Jadler contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package net.jadler.stubbing.server.jdk;
+
+import com.sun.net.httpserver.HttpServer;
```
Comment (liry) [inline jadler-jdk/src/main/java/net/jadler/stubbing/server/jdk/JdkStubHttpServer.java]: Does `com.sun...` package mean, it's not in all JDK implementations?

Comment (lukas-krecan) [inline jadler-jdk/src/main/java/net/jadler/stubbing/server/jdk/JdkStubHttpServer.java]: I can't find it anywhere (and I do not want to register to IBM just to download their JDK). But even if it were Oracle SDK only, it's an optional module so people do not have to use it.

**Comment by lukas-krecan-gdc** [comment]
Will you merge it? Alternatively, I can release it as a separate module.

**Comment by lukas-krecan** [comment]
I have released it as a separate module https://github.com/lukas-krecan/jadler-jdk (I wanted to use it) but I will be happy to kill the module once this PR is merged.

**Comment by liry** [comment]
@jandudek please review it and merge if it's ok (I don't have push permission)

File: jadler-jdk/src/test/java/net/jadler/stubbing/server/jdk/JdkStubHttpServerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import static net.jadler.Jadler.initJadlerUsing;
+import static net.jadler.Jadler.onRequest;
+import static net.jadler.Jadler.port;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests that in works without jadler-all.
+ */
+public class JdkStubHttpServerIntegrationTest {
```
Comment (jandudek) [inline jadler-jdk/src/test/java/net/jadler/stubbing/server/jdk/JdkStubHttpServerIntegrationTest.java]: What exactly does this test tests?

Comment (lukas-krecan-gdc) [inline jadler-jdk/src/test/java/net/jadler/stubbing/server/jdk/JdkStubHttpServerIntegrationTest.java]: I do not remember any more. If we have the test in jadler-all, we can remove this one.

Comment (lukas-krecan) [inline jadler-jdk/src/test/java/net/jadler/stubbing/server/jdk/JdkStubHttpServerIntegrationTest.java]: Shall I remove it?

File: jadler-jdk/src/main/java/net/jadler/stubbing/server/jdk/JdkStubHttpServer.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Stub server implementation based on {@link HttpServer} which is part of JDK.
+ */
+public class JdkStubHttpServer implements StubHttpServer {
+    private final HttpServer server;
+
+    public JdkStubHttpServer(int port) {
+        try {
+            server = HttpServer.create(new InetSocketAddress(port), 100);
```
Comment (jandudek) [inline jadler-jdk/src/main/java/net/jadler/stubbing/server/jdk/JdkStubHttpServer.java]: Do you remember what is the reason for the `100` constant? What about using 0 to enforce the system default value?

File: jadler-jdk/src/test/java/net/jadler/stubbing/server/jdk/JdkStubHttpServerIntegrationTest.java (hunk 4 above)
Comment (jandudek) [inline jadler-jdk/src/test/java/net/jadler/stubbing/server/jdk/JdkStubHttpServerIntegrationTest.java]: No, I will do it. I have forked your repo, doing it right now.

File: jadler-jdk/src/main/java/net/jadler/stubbing/server/jdk/JdkStubHttpServer.java (hunk 5 above)
Comment (lukas-krecan-gdc) [inline jadler-jdk/src/main/java/net/jadler/stubbing/server/jdk/JdkStubHttpServer.java]: Most likely copy and paste. 0 should be fine

**Comment by jandudek** [comment]
merged in my pull request and released as 1.2.0 to the public mvn repo

**Pull request merged**